Frends.Community.Email.Tests/ExchangeEmailTests.cs

```diff
@@ -0,0 +1,651 @@
+        {
+            var subject = "One Email Test";
+            await SendTestEmail(subject);
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 1,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = false,
+                MarkEmailsAsRead = false,
+                IgnoreAttachments = true,
+                EmailSubjectFilter = subject
+            };
+
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.That(result.Count, Is.EqualTo(1));
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task ReadEmailFromExchangeServer_ShouldReadFiveItemsTest()
+        {
+            var subject = "Five Emails test";
+            for (int i = 0; i < 5; i++)
+                await SendTestEmail(subject);
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 5,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = false,
+                MarkEmailsAsRead = false,
+                IgnoreAttachments = true,
+                EmailSubjectFilter = subject
+            };
+
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.That(result.Count, Is.EqualTo(5));
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task ReadEmailFromExchangeServer_ShouldGetUnreadMailsTest()
+        {
+            var subject = "Get Unread Emails Test";
+            await SendTestEmail(subject);
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 1,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = true,
+                MarkEmailsAsRead = true,
+                IgnoreAttachments = true,
+                EmailSubjectFilter = subject
+            };
+
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(result.Count == 1);
```

Comment:
Please use Assert equal instead. Because if this assert fails then error message is "Assert is true failed", which is very non-informative. But the equals assert gives a much better idea of what went wrong. Like "Assert equals fails, expected value ABC and god XYZ", which is quite helpful.